tools/build/cbt/fs.js

```diff
@@ -7,87 +7,80 @@
 const fs = require('fs');
 const glob = require('./glob');
 
-const Flags = {
-  SOURCE: 1,
-  TARGET: 1 << 1,
-  FILE: 1 << 2,
-  GLOB: 1 << 3,
-};
-
 class File {
   constructor(path) {
     this.path = path;
-    this.stat = null;
   }
 
-  performStat() {
-    this.stat = fs.statSync(this.path);
+  get stat() {
+    if (this._stat === undefined) {
+      this._stat = stat(this.path);
+    }
+    return this._stat;
+  }
+
+  exists() {
+    return this.stat !== null;
   }
 
   get mtime() {
-    if (!this.stat) {
-      this.performStat();
+    return this.stat && this.stat.mtime;
```

Comment:
this isn't a boolean return value?

Reply:
Yes, this isn't a boolean return value. It's an equivalent of me doing `this.stat?.mtime`, but this syntax is not supported on node 12.

Reply:
JS dum